tools/bumper-bot/parser/parser.test.go

```diff
@@ -0,0 +1,46 @@
+package parser
+
+import (
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("Testing internal parser functions", func() {
+	Describe("ParseComponentsYaml function", func() {
+		Context("When yaml file is empty", func() {
+			It("should return an error because the unmarshaling failed", func() {
+			})
+		})
+		Context("When yaml file is corrupted", func() {
+			It("should return an error because the unmarshaling failed", func() {
+			})
+		})
+		Context("When yaml file is valid", func() {
+			It("should correctly parse entire file", func() {
+			})
+			It("should correctly specific params of specific component", func() {
+			})
+		})
+	})
+	Describe("UpdateComponentsYaml function", func() {
+		Context("When yaml file is empty", func() {
+			Context("and we try to write valid data to it", func() {
+				It("should NOT return an error because data is valid", func() {
+				})
+			})
+			Context("and we try to write corrupted data to it", func() {
+				It("should return an error because marshaling should fail", func() {
+				})
+			})
+		})
+		Context("When yaml file is corrupted", func() {
+			It("should return an error because there is nothing to validate", func() {
+			})
+		})
+		Context("When yaml file is valid", func() {
+			It("should correctly parse entire file", func() {
+			})
+			It("should correctly parse specific params of specific component", func() {
+			})
+		})
+	})
```

Comment:
What about tests that Update function really updates?

Reply:
added